R/plotBeta.R

```diff
@@ -30,141 +30,91 @@ plotBeta <- function(alpha, beta) {
+                         Stop_cut = 0.1, # a poor improvement threshold
+                         shade = TRUE, # paint the two areas under the curve
+                         note = TRUE) { # show values of the colored area
+  diff <- seq(from = -1, to = 1, length = 1000)
+  data <- data.frame(
+    grid = diff,
+    density = dbetadiff(z = diff, parY = parY, parX = parX)
   )
+  data$Stop <- ifelse(diff > -1 & diff < Stop_cut, TRUE, FALSE)
+  data$Go <- ifelse(diff > Go_cut & diff < 1, TRUE, FALSE)
 
-  graphics::axis(
-    side = 1, at = xticks,
-    labels =
-      paste(ifelse(xticks >= 0, "+", ""),
-        xticks * 100, "%",
-        sep = ""
-      )
+  Go_auc <- integrate(
+    f = dbetadiff,
+    parY = parY,
+    parX = parX,
+    lower = Go_cut, # Calculate probability of Go, if difference was at least `Go_cut`.
+    upper = 1
+  )
+  Stop_auc <- integrate(
+    f = dbetadiff,
+    parY = parY,
+    parX = parX,
+    lower = -1,
+    upper = Stop_cut # Calculate probability of Stop, if difference was at most `Stop_cut`.
   )
 
-  ## now color the go / stop prob areas
-
-  if (shade == 1) {
-    ## first stop:
-    stopGrid <- grid[grid <= cut_W]
-    nStop <- length(stopGrid)
-
-    graphics::polygon(
-      x =
-        c(
-          stopGrid,
-          rev(stopGrid)
-        ),
-      y =
-        c(
-          rep(0, nStop),
-          dbetadiff(rev(stopGrid), parY = parY, parX = parX)
-        ),
-      col = "red"
-    )
-
-    A_value <- stats::integrate(
-      f = dbetadiff,
-      parY = parY,
-      parX = parX,
-      lower = -1,
-      upper = cut_W
-    )
-    if (note == 1) {
-      graphics::mtext(
-        paste("Prob(diff< ", round(cut_W * 100), "%)=",
-          sprintf("%1.2f%%", 100 * as.numeric(A_value$value)),
-          sep = ""
-        ),
-        side = 2, line = 1, las = 1, cex = 1
-      )
-    }
-
-    ## then go:
-    goGrid <- grid[grid >= cut_B]
-    nGo <- length(goGrid)
-
-    graphics::polygon(
-      x =
-        c(
-          goGrid,
-          rev(goGrid)
-        ),
-      y =
-        c(
-          rep(0, nGo),
-          dbetadiff(rev(goGrid), parY = parY, parX = parX)
-        ),
-      col = "green"
-    )
-
-    B_value <- stats::integrate(
-      f = dbetadiff,
-      parY = parY,
-      parX = parX,
-      lower = cut_B,
-      upper = 1
-    )
+  Go_label <- paste("P(Go) is", round(Go_auc$value * 100, digits = 2), "%")
+  Stop_label <- paste("P(Stop) is", round(Stop_auc$value * 100, digits = 2), "%")
+  plot_title <- paste("According to Beta difference density", Go_label, "and", Stop_label)
 
-    if (note == 1) {
-      graphics::mtext(
-        paste(
-          sprintf("%1.2f%%", 100 * as.numeric(B_value$value)),
-          "=Prob(diff> ",
-          round(cut_B * 100), "%)",
-          sep = ""
-        ),
-        side = 4,
-        line = 1,
-        las = 1,
-        cex = 1
-      )
-    }
+  if (shade == TRUE) {
```

Comment:
```suggestion
  if (shade) {
```
with flag variables best just like this